qtconsole/frontend_widget.py

```diff
@@ -233,14 +229,18 @@ def _is_complete(self, source, interactive):
             prompt created. When triggered by an Enter/Return key press,
             'interactive' is True; otherwise, it is False.
         """
-        self._input_splitter.reset()
-        try:
-            complete = self._input_splitter.push(source)
-        except SyntaxError:
-            return True
-        if interactive:
-            complete = not self._input_splitter.push_accepts_more()
-        return complete
+        kc = self.blocking_client
```

Comment:
Your `.blocking_client` property has a code path that will return None. This code won't handle that.